python/tests/integration/arcticdb/test_read_batch_more.py

```diff
@@ -0,0 +1,364 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+from arcticdb import DataError, ErrorCode
+from arcticdb.version_store.library import ReadRequest
+from arcticdb.version_store.processing import QueryBuilder
+import pytest
+import numpy as np
+import pandas as pd
+import datetime as dt
+import re
+
+from typing import Any
+from arcticdb.util.test import (assert_frame_equal, 
+                                create_df_index_datetime, 
+                                dataframe_simulate_arcticdb_update_static, 
+                                get_sample_dataframe,
+                                assert_frame_equal_row_range_fix,
+                                dataframe_single_column_string,
+                                dataframe_combine_all_dfs_into_one,
+                                dataframe_filter_with_datetime_index
+                                )
+
+def dataframe_concat_sort(*df_args : pd.DataFrame) -> pd.DataFrame:
+    """
+        Concatenates and sorts row range indexed dataframes
+    """
+    result = pd.concat(df_args[0:2], copy=True)
+    for idx in range(2,len(df_args),1):
+        result = pd.concat([result, df_args[idx]],copy=True)
+    result.sort_index(inplace=True) # We need to sort it at the end
+    return result
+
+def test_read_batch_2tables_7reads_different_slices(arctic_library):
+    """
+        Test aims to check if combined read of couple of DF, with several 
+        reads from each, which filters different subsections of the timeframes
+        is correct, in other words each read request is isolated from each other.
+        Covers columns, as_of and date_range parameters of read_batch() function
+    """
+    lib = arctic_library
+
+    symbol1 = "sym1"
+    df1_0 = create_df_index_datetime(7, 0, 5)
```

Comment:
I think it will be more readable if we define the variables near the palace they are used versus the C-98 style of having all definitions at the top.